src/Biryuchkov/Problem_2/mainwindow.cpp

```diff
@@ -0,0 +1,43 @@
+#include "mainwindow.hpp"
+#include "ui_mainwindow.h"
+
+MainWindow::MainWindow(QWidget *parent) :
+    QMainWindow(parent),
+    ui(new Ui::MainWindow),
+    colorDialog(new QColorDialog(this))
+{
+    ui->setupUi(this);
+    colorDialog->setOption(QColorDialog::NoButtons);
+
+    QObject::connect(colorDialog, &QColorDialog::currentColorChanged,
+                     ui->cubeWidget, &CubeWidget::setColor);
+}
+
+MainWindow::~MainWindow()
+{
+    delete ui;
+}
+
+void MainWindow::on_facesButton_clicked()
+{
+    ui->cubeWidget->setFaceMode();
+    colorDialog->show();
+}
+
+void MainWindow::on_edgesButton_clicked()
+{
+    ui->cubeWidget->setEdgeMode();
+    colorDialog->show();
+}
+
+void MainWindow::on_faceNumSpinbox_valueChanged(int arg1)
```

Comment:
```suggestion
void MainWindow::on_faceNumSpinbox_valueChanged(const int arg1)
```
+ давайте дадим осмысленное имя